django/contrib/admindocs/views.py

```diff
@@ -353,8 +355,14 @@ def get_context_data(self, **kwargs):
             # Non-trivial TEMPLATES settings aren't supported (#24125).
             pass
         else:
-            # This doesn't account for template loaders (#24128).
-            for index, directory in enumerate(default_engine.dirs):
+            directories = set(default_engine.dirs)
+            for loader in default_engine.template_loaders:
+                if isinstance(loader, Loader):
+                    directories.update(loader.get_dirs())
+                elif isinstance(loader, Cached_Loader):
```

Comment:
Perhaps the `Cached_Loader` should have a `get_dirs()` method? Special casing it here doesn't look like a good separation of concerns (admindocs knowing about internals of the template system).